pyproject.toml

```diff
@@ -34,8 +34,9 @@ build-backend = "pdm.backend"
 [tool.pdm.dev-dependencies]
 dev = [
     "pytest>=7.4.3",
+    "pytest-asyncio~=0.23.6",
     # "pytest-asyncio~=0.20.3",
-    "pytest-asyncio @ https://github.com/joeblackwaslike/pytest-asyncio/releases/download/v0.20.4.dev42/pytest_asyncio-0.20.4.dev42-py3-none-any.whl",
+    # "pytest-asyncio @ https://github.com/joeblackwaslike/pytest-asyncio/releases/download/v0.20.4.dev42/pytest_asyncio-0.20.4.dev42-py3-none-any.whl",
```

Comment:
this forked dependency is essential. pytest-asyncio has a longstanding bug they refuse to fix with context propagation which breaks async fixtures in pytest.  The problem was that async fixtures that yield were having their finalizers run in a separate asyncio task that did not preserve contextvars.

I was able to fork this pytest plugin and fix that, and for the first time ever I saw async generator fixtures work under pytest.  It felt magical.  The upstream repo is not interested in solving the problem in this way, they would rather not solve it.  

The last time I checked, my fork was the only version of pytest-asyncio that addresses this bug, and without it the tests fail due to an async fixture not propagating its context.  Feel free to review my changes in the following repo: https://github.com/joeblackwaslike/pytest-asyncio